File: test_regex.py
import re

pr_ref = "puemos/hls-downloader/490"
pattern = r"^(?:https://github\.com/)?([^/]+)/([^/]+)/pull/(\d+)$|(?:([^/]+)/([^/#]+))?#?(\d+)$"
re_obj = re.compile(pattern)
match = re_obj.search(pr_ref)

if match:
    print(f"Match found!")
    for i, group in enumerate(match.groups()):
        print(f"Group {i+1}: {group}")
else:
    print("No match found")

File: test_regex.rs
use regex::Regex;

fn main() {
    let pr_ref = "puemos/hls-downloader/490";
    let pattern = r"^(?:https://github\.com/)?([^/]+)/([^/]+)/pull/(\d+)$|(?:([^/]+)/([^/#]+))?#?(\d+)$";
    let re = Regex::new(pattern).unwrap();

    if let Some(caps) = re.captures(pr_ref) {
        println!("Match found!");
        for i in 0..caps.len() {
            println!("Group {}: {:?}", i, caps.get(i).map(|m| m.as_str()));
        }
    } else {
        println!("No match found");
    }
}
